onchain/dusd/Lib.hs

```diff
@@ -0,0 +1,93 @@
+module Lib (getNextOutputByNft, checkAdminSig) where
+
+import Plutarch.Api.V2 (
+  PDatum (PDatum),
+  POutputDatum (POutputDatum),
+  PPubKeyHash,
+  PScriptPurpose (PSpending),
+  PTxInInfo (..),
+  PTxOut (..),
+ )
+import Plutarch.Prelude
+
+import Plutarch.Api.V1.Value (PCurrencySymbol, PTokenName, PValue)
+
+import Plutarch.Extensions.Data (ptryFromData)
+import Plutarch.Extensions.Monad (pletFieldC)
+import Plutarch.Extra.TermCont (pguardC, pletC, pletFieldsC, pmatchC)
+
+import GHC.Records (HasField)
+import Plutarch.Api.V1 (PMap, PValue (PValue))
+import Plutarch.Api.V1 qualified as Value
+import Plutarch.Api.V1.AssocMap qualified as AssocMap
+
+getNextOutputByNft ::
```

Comment:
Can you include a doc block here explaining what this does? I had to look at it a few times to understand exactly what it's doing. Real nice function though.